### ISSUE TITLE
Error in gradient of conv2D: dilation rates greater than 1 are not yet supported in gradients

### ISSUE THREAD

**Issue by bubai567** [tensorflow/tfjs#7906]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow.js installed from (npm or script link): tfjs-node
- TensorFlow.js version (use command below): 4.10.0
- Browser version:
- Tensorflow.js Converter Version:

I encountered an issue while working with the tf.layers.conv1d layer in my deep learning project. When attempting to apply **dilationRate greater than 1**, an error is raised with the message: "Error in gradient of conv2D: dilation rates greater than 1 are not yet supported in gradients.

Error in gradient of conv2D: dilation rates greater than 1 are not yet supported in gradients. Got dilations '1,4'

        let network = tf.sequential();
		
		
        
        network.add(tf.layers.inputLayer({
            inputShape: this.Shape,// this.timePortion, indlen 5days of closing price inside how many individual single value 2
        }));
		
		

		
		
		network.add(tf.layers.timeDistributed({
			layer: tf.layers.conv1d({
				filters: 16,
				kernelSize: 3,
			}),
		}));

		network.add(tf.layers.timeDistributed({
			layer: tf.layers.leakyReLU({
				alpha: 0.1
			})
		}));
		
		

		network.add(tf.layers.timeDistributed({
			layer: tf.layers.conv1d({
				filters: 32,
				kernelSize: 3,
			}),
		}));

		network.add(tf.layers.timeDistributed({
			layer: tf.layers.leakyReLU({
				alpha: 0.1
			})
		}));

		
		network.add(tf.layers.timeDistributed({
			layer: tf.layers.averagePooling1d({
				poolSize: 2,
			}),
		}));
		
		
		

		
		
		
		network.add(tf.layers.timeDistributed({
			layer: tf.layers.flatten(),
		}));

		// time series

		network.add(tf.layers.conv1d({
			filters: 150,
			kernelSize: 2,
			padding: 'valid',
			dilationRate: 1,
			strides: 1
		}));

		network.add(tf.layers.leakyReLU({
			alpha: 0.1
		}))

		network.add(tf.layers.conv1d({
			filters: 150,
			kernelSize: 2,
			padding: 'valid',
			dilationRate: 2,
			strides: 1
		}));

		network.add(tf.layers.leakyReLU({
			alpha: 0.1
		}))

		network.add(tf.layers.conv1d({
			filters: 150,
			kernelSize: 2,
			padding: 'same',
			dilationRate: 4,
			strides: 1
		}));

		network.add(tf.layers.leakyReLU({
			alpha: 0.1
		}))

		

		network.add(tf.layers.maxPooling1d({
			poolSize: 2,
		}));

		
		

		network.add(tf.layers.flatten());

		
		network.add(tf.layers.dense({
			units: 256,
		}));

		network.add(tf.layers.leakyReLU({
			alpha: 0.1
		}))
		
		network.add(tf.layers.dropout({ rate: 0.25 }));
		
		network.add(tf.layers.dense({
			units: 128,
		}));

		network.add(tf.layers.leakyReLU({
			alpha: 0.1
		}))

**Comment by gaikwadrahul8** [comment]
Hi, @bubai567 

Thank you for bringing this issue to our attention and if I'm not wrong currently we do not support `dilationRate` value greater than 1 and as per official documentation of [tf.layers.conv2d](https://js.tensorflow.org/api/latest/#layers.conv2d), `Currently, specifying any dilationRate value != 1 is incompatible with specifying any strides value != 1.`

`dilationRate` value and `strides` value can be equal without any error in the `tf.layers.conv2d `operation in TensorFlow.js, if I'm not wrong. Thank you!
